pkgs/by-name/ni/nixos-rebuild-ng/src/nixos_rebuild/__init__.py

```diff
@@ -131,47 +140,82 @@ def parser_warn(msg: str) -> None:
     if args.no_ssh_tty:
         parser_warn("--no-ssh-tty deprecated, SSH's TTY is never used anymore")
 
+    # TODO: use deprecated=True in Python >=3.13
+    if args.no_build_nix:
+        parser_warn("--no-build-nix deprecated, we do not build nix anymore")
+
     if args.action == Action.EDIT.value and (args.file or args.attr):
         parser.error("--file and --attr are not supported with 'edit'")
 
-    if args.target_host and args.action not in (
+    if (args.target_host or args.build_host) and args.action not in (
         Action.SWITCH.value,
         Action.BOOT.value,
         Action.TEST.value,
         Action.BUILD.value,
         Action.DRY_BUILD.value,
         Action.DRY_ACTIVATE.value,
     ):
-        parser.error(f"--target-host is not supported with '{args.action}'")
+        parser.error(
+            f"--target-host/--build-host is not supported with '{args.action}'"
+        )
 
     if args.flake and (args.file or args.attr):
         parser.error("--flake cannot be used with --file or --attr")
 
-    if args.help or args.action is None:
-        r = run(["man", "8", "nixos-rebuild"], check=False)
-        parser.exit(r.returncode)
-
     return args, args_groups
 
 
+def reexec(
+    argv: list[str],
+    args: argparse.Namespace,
+    build_flags: dict[str, Args],
+    flake_build_flags: dict[str, Args],
+) -> None:
+    drv = None
+    try:
+        # Need to set target_host=None, to avoid connecting to remote
+        if flake := Flake.from_arg(args.flake, None):
+            drv = nix.build_flake(
+                "pkgs.nixos-rebuild-ng",
```

Comment:
I think this should eventually be `config.system.build.nixos-rebuild`, so that we get a nixos-rebuild that's built against `config.nix.package`. 

Maybe we could introduce `config.system.build.nixos-rebuild-ng` for now?

Currently, it's not really possible to get the right nixos-rebuild-ng in all situations, since from a generic overlay, you can't always select the right nix derivation to inject.

Reply:
Yes, I will eventually introduce `config.system.build.nixos-rebuild-ng` when I change the module system to make an opt-in for `nixos-rebuild-ng`.

For now this is mostly for testing.

Reply:
> Currently, it's not really possible to get the right nixos-rebuild-ng in all situations, since from a generic overlay, you can't always select the right nix derivation to inject.

But you make a good point, I will disable this for now.